src/vos/vos_container.c

```diff
@@ -459,8 +466,12 @@ vos_cont_open(daos_handle_t poh, uuid_t co_uuid, daos_handle_t *coh)
 		DP_UUID(cont->vc_id));
 
 exit:
-	if (rc != 0 && cont)
-		cont_decref(cont);
+	if (rc != 0 && cont) {
+		if (cont->vc_open_count > 0)
```

Comment:
How this can happen? If the container is just added into the cache, its vc_open_count is 1 set at line 464, but at that time "rc" is zero. Another case is that the container is not new, it is in cache already, set at 356, the "rc" is zero from such branch.

Reply:
yes, you are right. vc_open_count will always be 0 in this case.  I actually thought about that before adding this. Just thought add checking here makes more sense in general. 
Just in case some one do more things after setting vc_open_count later.

Reply:
it's a pretty standard programming way that we just call a free() instead of decref() in initialization function.